### PR TITLE
New resource specification for EnvoyConfig custom resources

### PR DESCRIPTION
**Pull request by roivaz**
This PR changes the way that resources are specified in the EnvoyConfig custom resource. A new field `spec.resources` has been added that is meant to replace `spec.envoyResources`. Both fields and specification formats work currently, with the controller internally transforming from `spec.envoyResources` to `spec.resources`.

The new resource specification looks like this:

```yaml
apiVersion: marin3r.3scale.net/v1alpha1
kind: EnvoyConfig
metadata:
  name: test
spec:
  nodeID: test
  resources:
    - type: listener
      value:
        name: https
        address:
          socket_address:
            address: 0.0.0.0
            port_value: 8443
        filter_chains:
          - filters:
              - name: envoy.filters.network.http_connection_manager
                typed_config:
                  "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
                  access_log:
                    - name: envoy.access_loggers.file
                      typed_config:
                        "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
                        path: /dev/stdout
                  stat_prefix: ingress_http
                  rds:
                    route_config_name: test
                    config_source:
                      ads: {}
                      resource_api_version: "V3"
                  http_filters:
                    - name: envoy.filters.http.router
                      typed_config:
                        "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
            transport_socket:
              name: envoy.transport_sockets.tls
              typed_config:
                "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext
                common_tls_context:
                  tls_certificate_sds_secret_configs:
                    - name: test
                      sds_config:
                        ads: {}
                        resource_api_version: "V3"
    - type: secret
      generateFromTlsSecret: test
    - type: route
      value:
        name: test
        virtual_hosts:
          - name: all
            domains: ["*"]
            routes:
              - match:
                  prefix: "/"
                route:
                  cluster: test
    - type: cluster
      value:
        name: test
        connect_timeout: 0.010s
        type: EDS
        lb_policy: ROUND_ROBIN
        eds_cluster_config:
          eds_config:
            ads: {}
            resource_api_version: "V3"
    - type: endpoint
      generateFromEndpointSlices:
        selector:
          matchLabels:
            kubernetes.io/service-name: test
        clusterName: test
        targetPort: http
```

Some other updates also in this PR, like upgrade of go-control-plane, controller-runtime, etc.

/kind feature
/priority important-soon
/assign

### CODE REVIEW

**Comment by roivaz** [comment]
/retitle New resource specification for EnvoyConfig custom resources

File: apis/marin3r/v1alpha1/resources.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+)
+
+// Resource holds serialized representation of an envoy
+// resource
+type Resource struct {
+	// Type is the type url for the protobuf message
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Enum=listener;route;scopedRoute;cluster;endpoint;secret;runtime;extensionConfig;
+	Type string `json:"type"`
+	// FromProto is the protobufer message that configures the resource. The proto
```
Comment (slopezz) [inline apis/marin3r/v1alpha1/resources.go:34]: Is this description correct for `Value`?

File: apis/marin3r/v1alpha1/resources.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	// The name of a Kubernetes Secret of type "kubernetes.io/tls"
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	GenerateFromTlsSecret *string `json:"generateFromTlsSecret,omitempty"`
+	// Specifies a label selector to watch for EndpointSlices that will
+	// be used to generate the endpoint resource
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	GenerateFromEndpointSlices *GenerateFromEndpointSlices `json:"generateFromEndpointSlices,omitempty"`
+	// Template specifies a template to generate a configuration proto. It is currently
```
Comment (slopezz) [inline apis/marin3r/v1alpha1/resources.go:49]: Is this description correct for `Blueprint`?

File: apis/marin3r/v1alpha1/resources.go (hunk 1 above)
Comment (roivaz) [reply apis/marin3r/v1alpha1/resources.go]: Description is accurate, but the field name changed several times and I forgot to update the description accordingly. Fixed with 7a3d23e

File: apis/marin3r/v1alpha1/resources.go (hunk 2 above)
Comment (roivaz) [reply apis/marin3r/v1alpha1/resources.go]: Description is accurate, but the field name changed several times and I forgot to update the description accordingly. Fixed with 7a3d23e

**Pull request merged**